### DISCUSSION TITLE
[Feature Suggestion] Package icons and customization features ?

### DISCUSSION THREAD

**Discussion by g2eat** [Devolutions/UniGetUI#788] [General]
Hi I had some ideas and was wondering they are already available, or if you would consider adding them in the future. 

1. Add Icons next to the packages like winstall, chocolatey and ninite.
So it would work like this, instead of the box Icon you already have, the package icon replace if the icon is available.  
It would be a useful feature because it much easier to find package with the icon rather than just  text. 

2.  Idea I had was have a section to show the type of app it is example Firefox - browser.

3. Short description of app. 

4 By (Publisher  or Author) - This would also give people peace of mind, its from a trustworthy source.

all the above features could be enabled and disabled in the setting giving everyone a choice. 
The Icons, short description and publisher are all already available when you click show info but it would be a little easy for a quick look and install option.

![icons 2](https://user-images.githubusercontent.com/5098039/224525836-5ab59d3d-0b26-460c-96d5-37546a6033e4.png)
![icons](https://user-images.githubusercontent.com/5098039/224525837-f785f2a9-64c2-47fe-b5dd-d8a52f2c51dd.png)

**Comment by marticliment** [comment]
This was discused earlier, but due lost of icons missing and the huge amount of packages shown, this would saturate the program and be way inefficient

**Comment by g2eat** [comment]
Thanks you for the reply 

Sorry I'm not a developer so  I thought it would be simpler, I assumed you used something similar to what is used by winstall to get the package data and  info. 

I dont know maybe you could try and speak to guy from winstall him via his  personal website,  I think he uses an API to do this maybe he would allow access to the API or  maybe work with you,  since the guy from winstall mostly focuses on the website side and you focus on a app,. 

Just a thought

**Reply by Spacellary** [reply]
Marti already tried formal contact in this https://github.com/MehediH/winstall/issues/129, it's yet to be responded...

**Reply by marticliment** [reply]
he didn't answer me neither, he might not be interested in collaborating...

**Comment by g2eat** [comment]
Thanks guys,  

I took a look at  that and a few others but I cant see any replies from him on any of them, the last thing I have seen from him was from 2021, either he has abandoned the project or just does not reply.  


Maybe a direct email would be better [meh@builtbymeh.com]
A last try to see if it possible

**Reply by Spacellary** [reply]
Emailing worked! He did reply to the issue (https://github.com/MehediH/winstall/issues/129), @marticliment!

:D

**Comment by g2eat** [comment]
Was this ever implemented ? A year on and I still can see the icons, unless its something on my end.

**Reply by marticliment** [reply]
Open package details or install a package, icons should show up

**Reply by g2eat** [reply]
You are correct some of them do show but what was the reason for not having the icons on the outside, like winstall does ?  
![Screenshot 2024-05-18 132151](https://github.com/marticliment/WingetUI/assets/5098039/3edb043e-72d6-4f98-aa81-180674508c94)

**Reply by marticliment** [reply]
There are not enough icons yet so that showing package icons is consistent and useful

**Reply by g2eat** [reply]
Ok, is that also the reason for not having the Icon on the outside, 
like this 
![Screenshot 2024-05-18 132151](https://github.com/marticliment/WingetUI/assets/5098039/d16c80ff-5577-413e-a40e-e1d111ccc244)
instead of this 
![Screenshot 2024-05-18 150521](https://github.com/marticliment/WingetUI/assets/5098039/cbcdae6d-6e9a-484b-9d0b-a7bd4cc72e8a)

also did the info from winstall ( MehediH ) not help to get the icons to get all the icons ?

**Reply by marticliment** [reply]
No, their Icon API is closed

**Reply by g2eat** [reply]
Ok, no issues, I thought it was open from the reply that was sent.

*the winstall API doesn't do anything fancy to for the icon engine - I simply use this npm library https://www.npmjs.com/package/get-website-favicon to get an app's website icon, then find the best resolution favicon file, and use that as the icon for the app.

this is then downloaded and saved in the winstall backend*

**Reply by marticliment** [reply]
Better icon integrations are on the works (both for WinGet and for all the NuGet-based package managers, so this will not be a requirement anymore)

**Reply by g2eat** [reply]
Looking forward to the future of the project.
